### PR TITLE
feat(ONYX-906): update required fields on sell flow

### PR DESCRIPTION
**Pull request by tam-kis**
The type of this PR is:  Feat

This PR solves [ONYX-906](https://artsyproduct.atlassian.net/browse/ONYX-906)

### Description

Update required fields for the `ArtworkDetailsForm` and add tracking for `width` and `height`

Optional width and height, `isSellFlowRequiredField` feature flag enabled:

https://github.com/artsy/force/assets/17580625/927362e1-4ffb-41a2-ba31-cbf8cfe063e7

Required width and height, `isSellFlowRequiredField` feature flag disabled:

https://github.com/artsy/force/assets/17580625/a8251a0f-350f-4fe5-8a83-07b4c2e2c221

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ONYX-906]: https://artsyproduct.atlassian.net/browse/ONYX-906?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Consign/Routes/SubmissionFlow/Utils/validation.ts [hunk 1]
```diff
@@ -7,20 +7,17 @@ export const artworkDetailsValidationSchema = yup.object().shape({
...
-    then: yup.string().required("Please enter the edition number").trim(),
+    then: yup.string().trim(),
   }),
   editionSize: yup.string().when("rarity", {
     is: "limited edition",
-    then: yup.string().required("Please enter the edition size").trim(),
+    then: yup.string().trim(),
   }),
   height: yup.string().required("Please enter the height").trim(),
   width: yup.string().required("Please enter the width").trim(),
```
Comment (tam-kis) [inline src/Apps/Consign/Routes/SubmissionFlow/Utils/validation.ts:23]: @dblandin I've tried somehow to add feature flag toggling here but unable to do so, any suggestions? I've left it just as required for now.

File: src/Apps/Consign/Routes/SubmissionFlow/Mutations/UpdateConsignSubmissionMutation.ts [hunk 2]
```diff
@@ -22,7 +23,7 @@ export const updateConsignSubmissionMutation = (
         }
       `,
       variables: {
-        input,
+        input: omitBy(input, isEmpty),
```
Comment (tam-kis) [inline src/Apps/Consign/Routes/SubmissionFlow/Mutations/UpdateConsignSubmissionMutation.ts:26]: @MounirDhahri was the decision to remove this in the end, suggestions how to resolve the issue?

File: src/Apps/Consign/Routes/SubmissionFlow/Utils/validation.ts (hunk 1 above)
Comment (dblandin) [reply src/Apps/Consign/Routes/SubmissionFlow/Utils/validation.ts:23]: A bit verbose but maybe something like this?

```tsx
{
  height: (isSellFlowRequiredField ? yup.string() : yup.string().required("Please enter the height")).trim(),
  width: (isSellFlowRequiredField ? yup.string() : yup.string().required("Please enter the height")).trim()
}
```

Comment (tam-kis) [reply src/Apps/Consign/Routes/SubmissionFlow/Utils/validation.ts:23]: I tried but we're not able to add the feature flag in this file unless we import it or refactor it entirely. 
I guess we could export the feature flag from ArtworkDetailsForm, where it's also used...

Comment (dblandin) [reply src/Apps/Consign/Routes/SubmissionFlow/Utils/validation.ts:23]: Ahh, I see. Yeah, in that case maybe we switch this helper to a generator function and update the few call sites that currently rely on `artworkDetailsValidationSchema`. Within the react component, we can check the feature flag and pass the boolean value in.

```ts

const baseArtworkDetailsValidationSchema = yup.object().shape({
  // snip...
  year: yup.string().trim(),
  // snip...
})

export const generateArtworkDetailsValidationSchema = (isSellFlowRequiredField: boolean = false) => {
  baseArtworkDetailsValidationSchema.concat(
    yup.object().shape({
      height: (isSellFlowRequiredField ? yup.string() : yup.string().required("Please enter the height")).trim(),
      width: (isSellFlowRequiredField ? yup.string() : yup.string().required("Please enter the height")).trim()
    })
  )
}
```

Comment (mzikherman) [reply src/Apps/Consign/Routes/SubmissionFlow/Utils/validation.ts:23]: Curious how come the feature flag isn't available? It should be available always within the codebase. There's a hook (which can't be used everywhere), but also the general `getENV`-backed fallback which should be accessible everywhere (client and server) - https://github.com/artsy/force/blob/6aefbee9677bdae053f6abd1828775584399079d/src/System/useFeatureFlag.tsx#L76

Comment (dblandin) [reply src/Apps/Consign/Routes/SubmissionFlow/Utils/validation.ts:23]: Ah, that's a good reminder about `getFeatureFlag` function. Makes sense that the React hook variant doesn't work in this context but we should try this lower-level function 👍🏼

File: src/Apps/Consign/Routes/SubmissionFlow/Mutations/UpdateConsignSubmissionMutation.ts (hunk 2 above)
Comment (MounirDhahri) [reply src/Apps/Consign/Routes/SubmissionFlow/Mutations/UpdateConsignSubmissionMutation.ts:26]: The issue in Eigen was that we were sending always all fields - this is fine in a flow that is uni-direction. It does however break if the user goes back and fires the mutation again, which can possibly reset already updated values. See this https://github.com/artsy/eigen/pull/10140#discussion_r1582833282

**Pull request merged**